sphinx/cmd/build.py

```diff
@@ -330,10 +338,17 @@ def build_main(argv: Sequence[str]) -> int:
                          args.tags, args.verbosity, args.jobs, args.keep_going,
                          args.pdb)
             app.build(args.force_all, args.filenames)
-            return app.statuscode
+            retcode = app.statuscode
     except (Exception, KeyboardInterrupt) as exc:
         handle_exception(app, args, exc, args.error)
-        return 2
+        retcode = 2
+
+    if warnfp is not None and closefd:
+        # close a file descriptor opened by Sphinx internally
+        # (only "second" stream of the Tee object can be safely
+        # closed since the first one may be sys.stderr).
+        warnfp.close()
+    return retcode
```

Comment:
I think it's cleaner to keep return and use finally here.

A

Reply:
Ah yes, thank you! I the `finally` clause is always something that I forgot to use 50% of the time.